### PR TITLE
feat(tc): support configure TiDB ssl-ca parameter via TiDBTLSClient

### LINKED ISSUE #4251: Support not setting `security.ssl-ca` when `spec.tidb.tlsclient.enabled: true`
## Feature Request

**Is your feature request related to a problem? Please describe:**

Currently, when the user set `spec.tidb.tlsclient.enabled: true` for TiDB, the following configurations will be set automatically [here](https://github.com/pingcap/tidb-operator/blob/v1.2.4/pkg/manager/member/tidb_member_manager.go#L388):
```
      [security]
        ssl-ca = "/var/lib/tidb-server-tls/ca.crt"
        ssl-cert = "/var/lib/tidb-server-tls/tls.crt"
        ssl-key = "/var/lib/tidb-server-tls/tls.key"
```

However, according to the doc [here](https://docs.pingcap.com/tidb/stable/enable-tls-between-clients-and-servers#configure-tidb-server-to-use-secure-connections), the `ssl-ca` is optional if the user does not need to enable the authentication between TiDB client and TiDB server.
So we have to support the above behavior.
**Describe the feature you'd like:**

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by just1900**
Signed-off-by: just1900 <legendj228@gmail.com>

### What problem does this PR solve?

Support configure TiDB ssl-ca parameter via TiDBTLSClient, Closes #4251 

### What is changed and how does it work?

- add a field called `DisableClientAuthn` to allow users configure the TiDB ssl-ca parameter. By default this field is set to false, so as to be compatible with current behavior(i.e. configure ssl-ca by default if TiDBTlsClient is enabled).
- add openapi-gen for TiDBTLSClient

### Code changes

- [x] Has Go code change

### Tests
- [x] Unit test 

### Side effects
no

### Related changes

- [x] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
Support configure TiDB ssl-ca parameter via TiDBTLSClient
```

### CODE REVIEW

File: pkg/apis/pingcap/v1alpha1/types.go [hunk 1]
```diff
@@ -1187,6 +1187,11 @@ type TiDBTLSClient struct {
 	//   4. Set Enabled to `true`.
 	// +optional
 	Enabled bool `json:"enabled,omitempty"`
+
+	// DisableClientAuthn will skip client authentication from the TiDB server.
+	// Optional: defaults to false
+	// +optional
+	DisableClientAuthn bool `json:"disableClientAuthn,omitempty"`
```
Comment (KanShiori) [inline pkg/apis/pingcap/v1alpha1/types.go:1194]: How about placing the field in class `TiDBTLSClient`?

**Review by KanShiori — APPROVED** [review]
LGTM

**Comment by KanShiori** [comment]
> [CLA assistant check](https://cla-assistant.io/pingcap/tidb-operator?pullRequest=4270) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/pingcap/tidb-operator?pullRequest=4270) before we can accept your contribution.You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/pingcap/tidb-operator?pullRequest=4270) it.

Please sign the CLA @just1900

File: pkg/manager/member/tidb_member_manager.go [hunk 2]
```diff
@@ -384,7 +384,10 @@ func getTiDBConfigMap(tc *v1alpha1.TidbCluster) (*corev1.ConfigMap, error) {
 		config.Set("security.cluster-ssl-key", path.Join(clusterCertPath, corev1.TLSPrivateKeyKey))
 	}
 	if tc.Spec.TiDB.IsTLSClientEnabled() {
-		config.Set("security.ssl-ca", path.Join(serverCertPath, tlsSecretRootCAKey))
+		// we won't configure the ssl-ca parameter when client authentication are disabled.
```
Comment (DanielZhangQD) [inline pkg/manager/member/tidb_member_manager.go:387]:
```suggestion
		// we won't configure the ssl-ca parameter when client authentication is disabled.
```

File: pkg/manager/member/tidb_member_manager.go [hunk 3]
```diff
@@ -384,7 +384,10 @@ func getTiDBConfigMap(tc *v1alpha1.TidbCluster) (*corev1.ConfigMap, error) {
...
+		if !tc.Spec.TiDB.TLSClient.DisableClientAuthn {
```
[the remark this answers is not in the document]
Comment (just1900) [reply pkg/manager/member/tidb_member_manager.go:388]: We have checked whether it is nil in `IsTLSClientEnabled()` func.

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Pull request merged**